src/streams.rs

```diff
@@ -51,9 +68,9 @@ impl BaseStream {
                 BaseStream::connect_tcp(host, port, info).map(|(stream, timeout)| BaseStream::Plain { stream, timeout })
             }
             #[cfg(feature = "tls")]
-            "https" => {
-                BaseStream::connect_tls(host, port, info).map(|(stream, timeout)| BaseStream::Tls { stream, timeout })
-            }
+            "https" => BaseStream::connect_tls(host, port, info),
+            #[cfg(feature = "tls-rustls")]
```

Comment:
Adding it this way might be surprising but allows one to build a functional instance of this crate with both features enabled which might happen if one depends on this indirectly with the dependencies in between enabling both features. Alternative one could force a compiler error when both features are enabled to prevent this situation from happening.

This will produce a warning about the pattern being unreachable in that case but should function normally preferring the existing `native-tls`-based code path. One could even consider the warning a feature but IIRC, Cargo hides warnings from dependencies by default...